File: src/main/java/com/refinedmods/refinedstorage/container/slot/legacy/LegacyDisabledSlot.java
package com.refinedmods.refinedstorage.container.slot.legacy;

import net.minecraft.inventory.IInventory;
import net.minecraft.item.ItemStack;

import javax.annotation.Nonnull;

public class LegacyDisabledSlot extends LegacyBaseSlot {
    public LegacyDisabledSlot(IInventory inventory, int inventoryIndex, int x, int y) {
        super(inventory, inventoryIndex, x, y);
    }

    @Override
    public boolean mayPlace(@Nonnull ItemStack stack) {
        return false;
    }
}

File: src/main/kotlin/com/refinedmods/refinedstorage/container/slot/legacy/LegacyDisabledSlot.kt
package com.refinedmods.refinedstorage.container.slot.legacy

import com.refinedmods.refinedstorage.api.network.grid.IGrid.slotId
import net.minecraft.inventory.IInventory
import net.minecraft.item.ItemStack

class LegacyDisabledSlot(inventory: IInventory?, inventoryIndex: Int, x: Int, y: Int) : LegacyBaseSlot(inventory, inventoryIndex, x, y) {
    fun isItemValid(@Nonnull stack: ItemStack?): Boolean {
        return false
    }
}